src/segment-loader.js

```diff
@@ -1328,4 +1562,144 @@ export default class SegmentLoader extends videojs.EventTarget {
 
     this.segmentMetadataTrack_.addCue(cue);
   }
+
+  // TODO: documentation
+  maxBufferedBytes(start, end) {
+    if (!this.segmentMetadataTrack_) {
+      return 0;
+    }
+
+    let i;
+    let j;
+    let cue;
+    let cmpStart;
+    let cmpEnd;
+    let bytes = 0;
+    let buffered = this.buffered_();
+
+    start = start === undefined ? 0 : start;
+    end = end === undefined ? Infinity : end;
+
+    for (i = 0; i < this.segmentMetadataTrack_.cues.length; i++) {
+      cue = this.segmentMetadataTrack_.cues[i];
+      for (j = 0; j < buffered.length; j++) {
+        cmpStart = (start > buffered.start(j)) ? start : buffered.start(j);
+        cmpEnd = (end < buffered.end(j)) ? end : buffered.end(j);
+        if (cue.startTime <= cmpEnd && cue.endTime >= cmpStart) {
+          bytes += cue.value.byteLength;
+          // we only want to count the bytes once, so break now
+          break;
+        }
+      }
+    }
+
+    return bytes;
+  }
+
+  // TODO: documentation
+  minBufferedBytes(start, end) {
+    if (!this.segmentMetadataTrack_) {
+      return 0;
+    }
+    if (end < start) {
+      return 0;
+    }
+
+    let i;
+    let j;
+    let cue;
+    let cmpStart;
+    let cmpEnd;
+    let bytes = 0;
+    let buffered = this.buffered_();
+
+    start = start === undefined ? 0 : start;
+    end = end === undefined ? Infinity : end;
+
+    for (i = 0; i < this.segmentMetadataTrack_.cues.length; i++) {
+      cue = this.segmentMetadataTrack_.cues[i];
+
+      for (j = 0; j < buffered.length; j++) {
+        cmpStart = start > buffered.start(j) ? start : buffered.start(j);
+        cmpEnd = end < buffered.end(j) ? end : buffered.end(j);
+
+        if (cue.endTime <= cmpEnd && cue.startTime >= cmpStart - 0.1) {
+          bytes += cue.value.byteLength;
+          break;
+        }
+      }
+    }
+
+    return bytes;
+  }
+
+  removeNonBufferedSegmentCues() {
+    if (!this.segmentMetadataTrack_) {
+      return 0;
+    }
+
+    let i;
+    let j;
+    let cue;
+    let cmpStart;
+    let cmpEnd;
+    let keep = true;
+    let removeCues = [];
+    let buffered = this.buffered_();
+
+    for (i = 0; i < this.segmentMetadataTrack_.cues.length; i++) {
+      cue = this.segmentMetadataTrack_.cues[i];
+      keep = false;
+
+      for (j = 0; j < buffered.length; j++) {
+        cmpStart = buffered.start(j);
+        cmpEnd = buffered.end(j);
+
+        if (cue.startTime <= cmpEnd && cue.endTime >= cmpStart) {
+          keep = true;
```

Comment:
couldn't you just call remove each cue as you decide to remove it? I think the `keep` flag is a bit confusing

Reply:
Ah, I think I meant to say couldn't you push a cue onto `removeCues` at this line?